client/my-sites/site-settings/site-visibility/launch-site.jsx

```diff
@@ -175,10 +176,10 @@ const LaunchSite = () => {
 										"Your site hasn't been launched yet. It's private; only you can see it until it is launched."
 								  ) }
 						</p>
-						{ isDevelopmentSite && <i>{ agencyBillingMessage }</i> }
+						{ isDevelopmentSite && ! siteReferralActive && <i>{ agencyBillingMessage }</i> }
 					</div>
 					<div className={ launchSiteClasses }>{ btnComponent }</div>
-					{ isDevelopmentSite && (
+					{ isDevelopmentSite && ! siteReferralActive && (
```

Comment:
Perhaps not displaying the Refer to Client button until the agency data is loaded would be better.
This means doing something like `!agencyLoading && isDevelopmentSite && ! siteReferralActive`.
Not a blocker.

Reply:
Nice catch! Thanks, Yan!

I have applied the change and also applied similar logic to the display of the billing message.

On top of that I made the `Launch site` button disabled if the agency data are still loading.